From 1cd8a8c7dbe3b5728caf68b6659fc59fe5b3031f Mon Sep 17 00:00:00 2001
From: Tomas Jelinek <tojeline@redhat.com>
Date: Wed, 31 May 2017 11:58:19 +0200
Subject: [PATCH] return 1 when pcsd is unable to bind

---
 pcsd/ssl.rb | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pcsd/ssl.rb b/pcsd/ssl.rb
index 24ee059..1a41ab2 100644
--- a/pcsd/ssl.rb
+++ b/pcsd/ssl.rb
@@ -99,8 +99,10 @@ def run_server(server, webrick_options, secondary_addrs)
   rescue Errno::EADDRNOTAVAIL, Errno::EADDRINUSE => e
     $logger.error 'Unable to bind to specified address(es), exiting'
     $logger.error e.message
+    exit 1
   rescue SocketError => e
     $logger.error e.message
+    exit 1
   end
 end
 
-- 
1.8.3.1

