From df10fbfd2673523f4cadac4be64cdf97ec9aba6c Mon Sep 17 00:00:00 2001
From: Tomas Jelinek <tojeline@redhat.com>
Date: Wed, 12 Aug 2015 15:47:09 +0200
Subject: [PATCH] improve logging in pcsd

---
 pcsd/pcs.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pcsd/pcs.rb b/pcsd/pcs.rb
index 6c7661a..1cddca8 100644
--- a/pcsd/pcs.rb
+++ b/pcsd/pcs.rb
@@ -359,6 +359,7 @@ def send_request_with_token(session, node, request, post=false, data={}, remote=
   token = additional_tokens[node] || get_node_token(node)
   $logger.info "SRWT Node: #{node} Request: #{request}"
   if not token
+    $logger.error "Unable to connect to node #{node}, no token available"
     return 400,'{"notoken":true}'
   end
   cookies_data = {
-- 
1.9.1

