From fc89908d91a2438f59dd08cf79aedfb85512091b Mon Sep 17 00:00:00 2001
From: Chris Feist <cfeist@redhat.com>
Date: Fri, 18 Sep 2015 16:29:58 -0500
Subject: [PATCH] Added more detailed warnings for 'pcs stonith confirm'

---
 pcs/pcs.8    | 4 +++-
 pcs/usage.py | 5 ++++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/pcs/pcs.8 b/pcs/pcs.8
index 70f0f6c..e89c813 100644
--- a/pcs/pcs.8
+++ b/pcs/pcs.8
@@ -333,7 +333,9 @@ fence <node> [\fB\-\-off\fR]
 Fence the node specified (if \fB\-\-off\fR is specified, use the 'off' API call to stonith which will turn the node off instead of rebooting it)
 .TP
 confirm <node>
-Confirm that the host specified is currently down.  WARNING: if this node is not actually down data corruption/cluster failure can occur.
+Confirm that the host specified is currently down.  This command should \fBONLY\fR be used when the node specified has already been confirmed to be down.
+
+.B WARNING: if this node is not actually down data corruption/cluster failure can occur.
 .SS "acl"
 .TP
 [show]
diff --git a/pcs/usage.py b/pcs/usage.py
index c430965..63baa76 100644
--- a/pcs/usage.py
+++ b/pcs/usage.py
@@ -833,7 +833,10 @@ Commands:
         call to stonith which will turn the node off instead of rebooting it)
 
     confirm <node>
-        Confirm that the host specified is currently down.
+        Confirm that the host specified is currently down.  This command
+        should ONLY be used when the node specified has already been
+        confirmed to be down.
+
         WARNING: if this node is not actually down data corruption/cluster
         failure can occur.
 
-- 
1.9.1

