From 2b843e7582ef3160a16094526727101180649448 Mon Sep 17 00:00:00 2001
From: Radek Steiger <rsteiger@redhat.com>
Date: Wed, 10 Aug 2016 09:56:28 +0200
Subject: [PATCH] fix detecting nonexisting resources in pcsd

---
 pcsd/remote.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pcsd/remote.rb b/pcsd/remote.rb
index 134ac5d..e467d0a 100644
--- a/pcsd/remote.rb
+++ b/pcsd/remote.rb
@@ -1584,7 +1584,7 @@ def remove_resource(params, request, auth_user)
         out, err, retval = run_cmd(user, *(cmd + [resource]))
         if retval != 0
           unless (
-            (out + err).join('').include?(' does not exist.') and
+            (out + err).join('').include?('unable to find a resource') and
             no_error_if_not_exists
           )
             errors += "Unable to stop resource '#{resource}': #{err.join('')}"
-- 
1.8.3.1

