From 01e0ceba838c558c08b11c51646af6e8d26a699b Mon Sep 17 00:00:00 2001
From: Ivan Devat <idevat@redhat.com>
Date: Wed, 27 Jul 2016 16:09:01 +0200
Subject: [PATCH] install bash completion with standard permissions

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index cbbeb85..25fb87d 100644
--- a/Makefile
+++ b/Makefile
@@ -85,7 +85,7 @@ install:
 	$(PYTHON) setup.py install --root=$(or ${DESTDIR}, /) ${EXTRA_SETUP_OPTS}
 	mkdir -p ${DESTDIR}${PREFIX}/sbin/
 	mv ${DESTDIR}${PREFIX}/bin/pcs ${DESTDIR}${PREFIX}/sbin/pcs
-	install -D pcs/bash_completion.sh ${BASH_COMPLETION_DIR}/pcs
+	install -D -m644 pcs/bash_completion.sh ${BASH_COMPLETION_DIR}/pcs
 	install -m644 -D pcs/pcs.8 ${DESTDIR}/${MANDIR}/man8/pcs.8
 ifeq ($(IS_DEBIAN),true)
   ifeq ($(install_settings),true)
-- 
1.8.3.1

